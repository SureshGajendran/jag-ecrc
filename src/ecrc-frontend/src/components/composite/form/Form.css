.ecrcForm {
  width: 100%;
  /* background-color: yellow; */
  overflow: hidden;
  padding-left: 20px;
  padding-right: 20px;
  padding-bottom: 10px;
}

ul {
  list-style-type: none;
  /* background: #3399ff; */
  padding: 20px;
  width: 100%;
}

ul li {
  margin: 15px;
}
