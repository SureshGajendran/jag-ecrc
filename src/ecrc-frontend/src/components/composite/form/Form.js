import React from "react";
import PropTypes from "prop-types";
import "./Form.css";
import TextInput from "../../base/textInput/TextInput";

export const Form = ({ textInputs }) => {
  const inputList = textInputs.map(input => {
    return (
      <li>
        <div>
          <TextInput key={input.name} textInput={input} />
        </div>
      </li>
    );
  });

  return (
    <form className="ecrcForm">
      {inputList.length > 0 && (
        <div className="row">
          <ul>{inputList}</ul>
        </div>
      )}
    </form>
  );
};
