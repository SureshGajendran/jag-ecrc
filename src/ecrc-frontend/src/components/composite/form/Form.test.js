/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { create } from "react-test-renderer";

import { Form } from "./Form";

describe("Form Component", () => {
  const textInputs = [
    {
      label: "label1",
      id: "textInput1",
      onChange: () => jest.fn()
    },
    {
      label: "label3",
      id: "textInput3",
      textInputStyle: "textinput_editable_gray",
      isRequired: true
    }
  ];

  const buttons = [
    {
      label: "button2",
      buttonStyle: "btn btn-warning",
      buttonSize: "btn btn-lg",
      type: "button",
      onClick: () => jest.fn()
    },
    {
      label: "button3",
      buttonStyle: "btn btn-danger",
      buttonSize: "btn btn-sm",
      type: "button",
      onClick: () => jest.fn()
    }
  ];

  test("Matches the snapshot", () => {
    const form = create(
      <Form title="title" textInputs={textInputs} buttons={buttons} />
    );
    expect(form.toJSON()).toMatchSnapshot();
  });
});
