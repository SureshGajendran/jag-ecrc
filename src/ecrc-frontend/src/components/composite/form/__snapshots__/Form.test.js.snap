// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component Matches the snapshot 1`] = `
<form
  className="ecrcForm"
>
  <div
    className="ecrcForm_title"
  >
    title
  </div>
  <ul>
    <li>
      <div>
        <div>
          <label
            className="textinput_label"
            htmlFor="textInput1"
          >
            label1
          </label>
          
        </div>
        <input
          className="textinput_editable_white"
          id="textInput1"
          onChange={[Function]}
          type="text"
        />
      </div>
    </li>
    <li>
      <div>
        <div>
          <label
            className="textinput_label"
            htmlFor="textInput3"
          >
            label3
          </label>
          <span
            className="mandatory"
            id="asterisk"
          >
            *
          </span>
        </div>
        <input
          className="textinput_editable_gray"
          id="textInput3"
          onChange={[Function]}
          type="text"
        />
      </div>
    </li>
  </ul>
  <ul
    id="buttonList"
  >
    <li>
      <button
        className="btn btn-warning btn btn-lg"
        onClick={[Function]}
        type="button"
      >
        button2
      </button>
    </li>
    <li>
      <button
        className="btn btn-danger btn btn-sm"
        onClick={[Function]}
        type="button"
      >
        button3
      </button>
    </li>
  </ul>
</form>
`;
